Testreglar/3.3.1/App/app-3.3.1b.json

```diff
@@ -0,0 +1,276 @@
+			"type": "jaNei",
+			"ruting": {
+				"ja": {
+					"type": "ikkjeForekomst",
+					"utfall": "Skjema oppdager ikke automatisk inndata som faller utenfor påkrevd dataformat eller verdier.",
+					"element": "2.3"
+				},
+				"nei": {
+					"type": "gaaTil",
+					"steg": "3.1"
+				}
+			}
+		},
+		{
+			"stegnr": "3.1",
+			"spm": "Hvilket skjemaelement tester du?",
+			"ht": "<p>Beskriv elementet, slik at det er mulig å identifisere det i ettertid. Hvis det gjelder flere elementer, registrerer du et og et.</p>",
+			"type": "tekst",
+			"label": "Skjemaelement:",
+			"multilinje": true,
+			"oblig": true,
+			"ruting": {
+				"alle": {
+					"type": "gaaTil",
+					"steg": "3.2"
+				}
+			}
+		},
+		{
+			"stegnr": "3.2",
+			"spm": "Blir informasjonen du har fylt ut liggende i skjemaelementet?",
+			"ht": "<p>Hovedregelen er at informasjon som er fylt ut skal bli liggende i skjemaet, selv om det blir oppdaget inndatafeil. Unntaket er dersom informasjonen er knyttet til personvern eller sikkerhet (sensitiv informasjon). </p>\n<p><strong>Merk:</strong> Det er ok at feil inndata blir korrigert automatisk, men inndatafeilen skal likevel beskrives til brukeren etter suksesskriterium 3.3.1.</p>",
+			"type": "jaNei",
+			"kilde": [
+				"G84",
+				"G85"
+			],
+			"ruting": {
+				"ja": {
+					"type": "gaaTil",
+					"steg": "3.3"
+				},
+				"nei": {
+					"type": "avslutt",
+					"fasit": "Nei",
+					"utfall": "I skjemaelement der inndatafeil blir oppdaget automatisk, forsvinner inndata fra skjemaelementet."
+				}
+			}
+		},
+		{
+			"stegnr": "3.3",
+			"spm": "Får du en tekstlig feilmelding?",
+			"ht": "<p>Feilmeldingen kan for eksempel vises</p>\n<ul>\n<li>i et modalvindu</li>\n<li>i ledeteksten til skjemaelement</li>\n<li>et annet sted i skjemaet/på siden, for eksempel på toppen</li>\n<li>når du navigerer i skjemaet</li>\n</ul>\n<p><strong>Merk:</strong> Det er ikke nok å identifisere feilen utelukkende med for eksempel </p>\n<ul>\n<li>et symbol i ledeteksten</li>\n<li>ved å endre farge</li>\n<li>visuell plassering av feilmeldingen</li>\n<li>deaktivert knapp for å sende inn eller gå videre i skjemaet</li>\n<li>lignende ikke-tekstlige indikatorer</li>\n</ul>",
+			"type": "jaNei",
+			"kilde": [
+				"G84",
+				"G85"
+			],
+			"ruting": {
+				"ja": {
+					"type": "gaaTil",
+					"steg": "3.4"
+				},
+				"nei": {
+					"type": "avslutt",
+					"fasit": "Nei",
+					"utfall": "Skjemaelement der inndatafeil blir oppdaget automatisk, får ikke tekstlig feilmelding."
+				}
+			}
+		},
+		{
+			"stegnr": "3.4",
+			"spm": "Er feilmeldingen kodet som tekst?",
+			"ht": "<p>Slik tester du:</p>\n<ul>\n<li>Aktiver skjermleser (opplesingsfunksjon)\n<ul>\n<li>iOS: VoiceOver</li>\n<li>Android: TalkBack</li>\n</ul>\n</li>\n<li>Trykk på feilmeldingen du tester, og sjekk om den blir lest opp</li>\n</ul>\n<p><strong>Merk:</strong> Hvis feilmeldingen ikke blir lest opp, er den ikke kodet som tekst.</p>",
```

Comment:
Trykk eller sveip til.

Reply:
Hvis feilmeldingen ikke blir lest opp, er den ikke kodet som tekst. -> Vi bør heller si at vi **antar** at den ikke er kodet som tekst. Det er rom for at det er andre forhold, feks bug i skjermleser eller feil i accessbilty API som er kan være årsak, og siden vi ikke sjekker koden vet vi ikke dette helt sikkert.